lcfs/diff.c

```diff
@@ -15,19 +15,22 @@ static void
 lc_addFile(struct fs *fs, struct cdir *cdir, ino_t ino, char *name,
            uint16_t len, enum lc_changeType ctype) {
     struct cfile *cfile = cdir->cd_file, **prev = &cdir->cd_file;
+    bool found = false;
 
     assert(cdir->cd_type != LC_REMOVED);
 
     /* Check if the file already in the list */
-    while (cfile &&
-           ((cfile->cf_len != len) || strncmp(cfile->cf_name, name, len))) {
+    while (cfile) {
+        if ((cfile->cf_len == len) && !strncmp(cfile->cf_name, name, len)) {
+            found = true;
```

Comment:
This variable found could be eliminated if the actual work is done here.

Reply:
en，But this code is more readable, and optimizations can be done by the compiler.